### DISCUSSION TITLE
3.0.0rc2: spacy download en_vectors_web_lg returns 404

### DISCUSSION THREAD

**Discussion by vitaly-d** [explosion/spaCy#6570] [Help: Coding & Implementations] [answered]
Hello!
For some reason `spacy download en_vectors_web_lg` returns 404
```
% spacy info

============================== Info about spaCy ==============================

spaCy version    3.0.0rc2
Location         /Users/vitaly/.venv/lib/python3.8/site-packages/spacy
Platform         macOS-10.16-x86_64-i386-64bit
Python version   3.8.5
Pipelines        en_core_web_lg (3.0.0a0), en_core_web_trf (3.0.0a0)
```

```
Collecting en_vectors_web_lg==3.0.0a0
  ERROR: HTTP error 404 while getting https://github.com/explosion/spacy-models/releases/download/en_vectors_web_lg-3.0.0a0/en_vectors_web_lg-3.0.0a0.tar.gz#egg=en_vectors_web_lg==3.0.0a0
ERROR: Could not install requirement en_vectors_web_lg==3.0.0a0 from https://github.com/explosion/spacy-models/releases/download/en_vectors_web_lg-3.0.0a0/en_vectors_web_lg-3.0.0a0.tar.gz#egg=en_vectors_web_lg==3.0.0a0 because of HTTP error 404 Client Error: Not Found for url: https://github.com/explosion/spacy-models/releases/download/en_vectors_web_lg-3.0.0a0/en_vectors_web_lg-3.0.0a0.tar.gz for URL https://github.com/explosion/spacy-models/releases/download/en_vectors_web_lg-3.0.0a0/en_vectors_web_lg-3.0.0a0.tar.gz#egg=en_vectors_web_lg==3.0.0a0
```

**Comment by svlandeg** [comment, marked as answer]
Hi! 

For v3, the `vectors` models are not distributed as a separate package anymore. See [here](https://nightly.spacy.io/models/en) for a list of available English models. If you want to have just the vectors, you can load `en_core_web_lg` and [disable or exclude](https://nightly.spacy.io/usage/processing-pipelines#disabling) all pipeline components. If you exclude them, and save the pipeline back to disk, you'll effectively have a model package with just the vectors.
```
nlp = spacy.load("en_core_web_lg", exclude=["ner", "tagger", "parser"])
nlp.to_disk("my_vectors")
```

**Reply by adrianeboyd** [reply]
Note that the vectors in `en_vectors_web_lg` and `en_core_web_lg` aren't identical. If you do want the exact same vectors for some reason, you can export them in word2vec format from the v2 model and then import them into a v3 model with `spacy init model`. There are a few missing strings and some weird whitespace entries in `en_vectors_web_lg`, so the `n_keys` count in the first row will slightly off, but `spacy init model` doesn't validate or use this count (another application might, though, so then you'd need to adjust it!):

```python
import spacy
nlp = spacy.load("en_vectors_web_lg")
with open("vectors.vec", "w") as fileh:
    fileh.write(str(nlp.vocab.vectors.n_keys) + " " + str(nlp.vocab.vectors.shape[1]) + "\n")
    for key in nlp.vocab.vectors.key2row:
        try:
            word = nlp.vocab.strings[key]
            if not word.isspace():
                fileh.write(word + " " + " ".join(str(x) for x in nlp.vocab.vectors[key]) + "\n")
        except KeyError:
            pass
```

These are based on the original glove vectors, which are getting a bit out-of-date, so you may well prefer to use newer vectors from another source.

**Comment by vitaly-d** [comment]
Thank you! 
I have another question . It might me better to start a new thread, but this one is also about migration to 3.0. We have a bunch of 2.x Spacy model in prod, probably the best way to migrate is just to re-train all models with 3.0. 
But as re-training is a bit time-consuming, I'm wondering would it be possible to 1) init a new 3.0 model with vectors as described above , 2) add ner/parser sections to the model config, and 3) copy 2.x "ner"/"parser" into the new model ?

It seems like the step 3) is most tricky one because I ended up with the "Trying to read a Model that was created with an incompatible version of Thinc" error.  And have read it as "it would not work for the obvious reason, instead all models needs to be re-trained with 3.0"  :) 
But then again, what if I'm in one step from a 2.x model that works in 3.0... Could you please share you thoughts?

P.S.
this issue is relevant to my question: [Functionality to copy trained Spacy NER + Parser model from Spacy2.x to Spacy3.x format. spacy-nightly · Issue #6330 · explosion/spaCy · GitHub](https://github.com/explosion/spaCy/issues/6330)
